plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java

```diff
@@ -380,6 +382,12 @@ public List<String> getAllBrokersForTable(String table)
 
     public String getBrokerHost(String table)
     {
+        // Use global broker URI if provided explicitly via config
+        if (brokerUri.isPresent()) {
+            URI uri = brokerUri.get();
+            return HostAndPort.fromParts(uri.getHost(), uri.getPort()).toString();
```

Comment:
No requested change. We build `HostAndPort` and change to `String` here and then revert to `HostAndPort` in `getBrokerHttpUriBuilder`. This is a redundant cast.